File: 824.goat-latin.py
#
# @lc app=leetcode id=824 lang=python3
#
# [824] Goat Latin
#

# @lc code=start
class Solution:
    def toGoatLatin(self, sentence: str) -> str:
        output = []
        vowels = ['A', 'E', 'I', 'O', 'U', 'a', 'e', 'i', 'o', 'u']
        for i,word in enumerate(sentence.split(' ')):
            if word[0] not in vowels:
                word = word[1::] + word[0]
            word = word + 'ma'

            word = word + 'a'*(i+1)
            output.append(word)
        return ' '.join(output)

# @lc code=end

File: js/824.goat-latin.js
/*
 * @lc app=leetcode id=824 lang=javascript
 *
 * [824] Goat Latin
 */

// @lc code=start
/**
 * @param {string} sentence
 * @return {string}
 */
var toGoatLatin = function(sentence) {
  return sentence.split(' ').map((v,i)=>{
    if (!['a','e','i','o','u','A','E','I','O','U'].includes(v[0]))
      v = v.slice(1) + v[0]
    v += 'ma'
    v += 'a'.repeat(i+1)
    return v
  }).join(' ')
};
// @lc code=end

